diff --git a/android-configure b/android-configure
index 59f0a40c1a..8813c043c1 100755
--- a/android-configure
+++ b/android-configure
@@ -17,6 +17,11 @@ fi
 
 CC_VER="4.9"
 case $ARCH in
+    arm64)
+        DEST_CPU="$ARCH"
+        SUFFIX="aarch64-linux-android"
+        TOOLCHAIN_NAME="$SUFFIX"
+        ;;
     arm)
         DEST_CPU="$ARCH"
         SUFFIX="$ARCH-linux-androideabi"
@@ -75,5 +80,7 @@ if [ -f "configure" ]; then
         --dest-cpu=$DEST_CPU \
         --dest-os=android \
         --without-snapshot \
-        --openssl-no-asm
+        --openssl-no-asm \
+        --shared \
+        --without-intl
 fi
